File: leetcode/27-RemoveElement/27-RemoveElement.cpp
// Last updated: 29/04/2026, 23:14:05
class Solution
{
public:
    int removeElement(vector<int> &nums, int val)
    {
        int swapper = 0;
        int pointer = 0;
        int temp = 0;
        int len = nums.size();
        while (pointer < len)
        {
            if (nums[swapper] == val)
            {
                if (nums[pointer] != val)
                {
                    // swap pointer and swapper
                    temp = nums[swapper];
                    nums[swapper] = nums[pointer];
                    nums[pointer] = temp;
                    swapper++;
                }
            }
            else
            {
                swapper++;
            }
            pointer++;
        }
        return swapper;
    }
};

File: leetcode/27-RemoveElement/27-RemoveElement.py
# Last updated: 29/04/2026, 23:14:00
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        l = 0
        r = 0
        while l < len(nums) and r < len(nums):
            if nums[r] != val:
                nums[l],nums[r] = nums[r],nums[l]
                l += 1
            r += 1
        return l